### PR TITLE
remove BadgerIterator.Released, simplify tests a bit

### PR DESCRIPTION
**Pull request by mvdan**

### CODE REVIEW

File: database_test.go [hunk 1]
```diff
@@ -263,45 +218,39 @@ func testSeekKeyValueIterator(db Database, t *testing.T) {
...
 
-	badgerDb, err := NewBadgerDB(dir)
+	db, err := NewBadgerDB(dir)
 	if err != nil {
 		t.Fatal(err)
 	}
-
-	db := Database(badgerDb)
-	return db, func() {
+	t.Cleanup(func() {
```
Comment (noot) [inline database_test.go:250]: this is very cool! I guess this from go 1.14? I'm running 1.13 right now, so I'll need to upgrade

**Review by noot — CHANGES_REQUESTED** [review]
looks good, could you update the README to say this requires go 1.14?

File: database_test.go (hunk 1 above)
Comment (mvdan) [reply database_test.go:250]: Yep! Will update the README, though this is a test requirement only.

**Review by noot — APPROVED** [review]

File: database_test.go (hunk 1 above)
Comment (noot) [reply database_test.go:250]: good point, maybe note that in the README

**Pull request merged**